src/systems/joint_controller/JointController.cc

```diff
@@ -85,23 +90,19 @@ void JointController::Configure(const Entity &_entity,
     return;
   }
 
+  // Ugly, but needed because the sdf::Element::GetElement is not a const
+  // function and _sdf is a const shared pointer to a const sdf::Element.
+  auto ptr = const_cast<sdf::Element *>(_sdf.get());
```

Comment:
CC: @azeey do you have any thoughts on how to tidy this up via sdformat?  It's probably out of the scope of this PR, but as @arjo129 said, it's a common pattern.

Reply:
Yes, it is now possible to just do `_sdf->FindElement` on a const `_sdf`.

Reply:
Hmmm... `FindElement` returns one element. I think the issue in this case is that we are looking for more than on element of with a given name. For instance I think this PR is trying to have multiple `<joint_name>` tags.

Reply:
Oh, I meant we should use `FindElement` in place of `GetElement`. The whole reason we needed to do `const_cast` is to be able to use `GetElement` because we didn't have `FindElement` in the past.  The rest of the loop including the call to `GetNextElement` is still necessary.

Reply:
In fact, `GetElement` is dangerous because it might create the element if it doesn't exist, so I'd consider the use of `GetElement` an anti-pattern for systems at this point. The only time it's useful is when you're generating an SDFormat output.

Reply:
> Hmmm... `FindElement` returns one element. I think the issue in this case is that we are looking for more than on element of with a given name. For instance I think this PR is trying to have multiple `<joint_name>` tags.

That's correct, I took the implementation from what I saw done in the Ackermann plugin already: https://github.com/gazebosim/gz-sim/blob/1eb3a9594c338e2d7f8e710b6909632930f8ead0/src/systems/ackermann_steering/AckermannSteering.cc#L219-L249

What might it look like with the now preferred `_sdf->FindElement` method?

Reply:
> In fact, `GetElement` is dangerous because it might create the element if it doesn't exist, so I'd consider the use of `GetElement` an anti-pattern for systems at this point. The only time it's useful is when you're generating an SDFormat output.

Is there a `FindNextElement` ? not sure the best way to implement that change.

Reply:
Here's an example of the preferred approach:
https://github.com/gazebosim/gz-sim/blob/1eb3a9594c338e2d7f8e710b6909632930f8ead0/src/LevelManager.cc#L204-L212

There is no `FindNextElement`. You'd still use `GetNextElement` which doesn't require a non-const `sdf::Element`. You can then remove the `const_cast` in the code.

Reply:
Done!